### PR TITLE
refactor: custom errors

### PR DESCRIPTION
**Pull request by r4mmer**
# Acceptance Criteria

Resolves @msbrogli review on #67

### CODE REVIEW

File: gateways/metadata_gateway.py [hunk 1]
```diff
@@ -44,12 +48,20 @@ def _get_metadata(self, s3_object_name: str) -> Optional[dict]:
         try:
             return json.loads(raw_metadata)
         except ValueError:
+            logging.warning('Metadata object {} is not a valid json'.format(s3_object_name))
```
Comment (pedroferreira1) [inline gateways/metadata_gateway.py:51]: Should this be `logger.warning`?

**Review by pedroferreira1 — CHANGES_REQUESTED** [review]

File: gateways/metadata_gateway.py (hunk 1 above)
Comment (r4mmer) [reply gateways/metadata_gateway.py]: Fixed

**Review by pedroferreira1 — APPROVED** [review]

**Review by luislhl — APPROVED** [review]

**Review by andreabadesso — APPROVED** [review]

File: gateways/token_gateway.py [hunk 2]
```diff
@@ -11,10 +11,10 @@ class TokenGateway:
     :param hathor_core_client: Client for make hathor-core requests, default to domain HathorCoreClient
     :type hathor_core_client:
     """
-    def __init__(self, hathor_core_client: Union[HathorCoreClient, None] = None) -> None:
+    def __init__(self, hathor_core_client: Optional[HathorCoreClient] = None) -> None:
         self.hathor_core_client = hathor_core_client or HathorCoreClient()
 
-    def get_token(self, id: str) -> Union[Token, None]:
+    def get_token(self, id: str) -> Optional[Token]:
```
Comment (msbrogli) [inline gateways/token_gateway.py:17]: Missing docstring.

File: utils/wrappers/aws/api_gateway.py [hunk 3]
```diff
@@ -68,5 +69,13 @@ def wrapper(event: dict, context: LambdaContext, *args: Any, **kwargs: Any) -> d
                         'error': str(error)
                     })
                 }
+            except Exception as error:
+                return {
```
Comment (msbrogli) [inline utils/wrappers/aws/api_gateway.py:73]: Shouldn't we log the error? I guess we will return 500 and just suppress the error itself.

File: utils/wrappers/aws/api_gateway.py [hunk 4]
```diff
@@ -55,7 +56,7 @@ def wrapper(event: dict, context: LambdaContext, *args: Any, **kwargs: Any) -> d
                 result['headers'].update(headers)
 
                 return result
-            except Exception as error:
+            except ApiError as error:
                 error_key = str(error)
```
Comment (msbrogli) [inline utils/wrappers/aws/api_gateway.py:60]: Same problem here. I guess we should log the exception.

**Review by msbrogli — APPROVED** [review]

File: utils/wrappers/aws/api_gateway.py (hunk 3 above)
Comment (r4mmer) [reply utils/wrappers/aws/api_gateway.py:73]: Added logs.

File: gateways/token_gateway.py (hunk 2 above)
Comment (r4mmer) [reply gateways/token_gateway.py:17]: Created.

**Pull request merged**